src/pipeline.cc

```diff
@@ -1320,6 +1348,24 @@ Napi::Value pipeline(const Napi::CallbackInfo& info) {
   baton->heifCompression = static_cast<VipsForeignHeifCompression>(
     vips_enum_from_nick(nullptr, VIPS_TYPE_FOREIGN_HEIF_COMPRESSION,
     sharp::AttrAsStr(options, "heifCompression").data()));
+
+  // Animated output
+  if (sharp::HasAttr(options, "pageHeight")) {
+    baton->pageHeight = sharp::AttrAsUint32(options, "pageHeight");
+  }
+  if (sharp::HasAttr(options, "loop")) {
+    baton->loop = sharp::AttrAsUint32(options, "loop");
+  }
+  if (sharp::HasAttr(options, "delay")) {
+    Napi::Array delay = options.Get("delay").As<Napi::Array>();
+    unsigned int delayLength = delay.Length();
+    baton->delay = new std::vector<int>(delayLength);
```

Comment:
The use of `new` here will leak memory. I think `baton->delay` can be a vector instance rather than a pointer - see `baton->resizeBackground` for an example.